### PR TITLE
AN-381: Polishing

### PR DESCRIPTION
**Pull request by aquarat**
Fixes based on @Siegrift's suggestions:
- consistent file naming
- remove unused/commented code
- setup precommit + CI checks
  - I copied Airnode's hooks and added prettier/eslint/solidity linters.
- ...and other polishing stuff

### CODE REVIEW

File: hardhat-build/hardhat.config.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+'use strict';
```
Comment (Siegrift) [inline hardhat-build/hardhat.config.js:1]: This seems like it should be ignored.

File: src/do_test_run.ts [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+    PostgresConfig,
+    JsonOutputConfig,
+    SshConfig,
+    TestType,
+    InfuraProviderAirnodeOverrideURL,
+    IgnoreNodeVersion,
+  } = getStressTestConfig();
+  const { SshKeyPath, SshRemoteHost, SshUser, SshPort, YamlPath } = SshConfig;
+
+  /*
```
Comment (Siegrift) [inline src/do_test_run.ts:49]: Strange formatting. Is the prettier setup correctly?

File: src/do_test_run.ts [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+      HardHat gets unhappy if you hit it with more than 10 RPC calls simultaneously.
+       */
+  const maxBatchSize = (() => {
```
Comment (Siegrift) [inline src/do_test_run.ts:52]: IIFEs are generally too unreadable, it's nicer to define a getter fn.

File: src/do_test_run.ts [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+        }
+      }
+      await refreshSecrets(RequestCount);
+
+      // Collect metrics
+      const collectedMetrics = await collectMetrics();
+      cliPrint.info(`Collected metrics: ${JSON.stringify(collectedMetrics, null, 2)}`);
+
+      return collectedMetrics;
+    })();
```
Comment (Siegrift) [inline src/do_test_run.ts:320]: This IIFE is unreadable.

**Review by Siegrift — APPROVED** [review]
Found a few minor readablity isssues, but otherwise LGTM.

Although the PR is very large :/

File: src/do_test_run.ts (hunk 2 above)
Comment (aquarat) [reply src/do_test_run.ts:49]: I will double-check - prettier's setup is a duplicate of Airnode's.

File: src/do_test_run.ts (hunk 3 above)
Comment (aquarat) [reply src/do_test_run.ts:52]: No problem, I'll split them all up. I'll take my cultural learnings from the EIP-1559 PR and apply them here :)

**Comment by aquarat** [comment]
> Found a few minor readablity isssues, but otherwise LGTM.
> 
> Although the PR is very large :/

Thanks for having a look :), the IIFEs were on my to-do list, I'm just balancing other tasks with this. I'll do smaller PRs in future.

File: .eslintignore [hunk 5]
```diff
@@ -0,0 +1,17 @@
+**/node_modules/**
+
+# Buidler
```
Comment (Siegrift) [inline .eslintignore:3]: Nobody uses buidler now. We use hardhat.

File: .eslintignore [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+**/.build/**
+**/.serverless/**
+**/coverage/**
+**/dist/**
+
+# Ignore the TS wrappers created by Typechain
+packages/airnode-protocol/src/contracts
+
+# Ignore JSON snippets injected into README
+packages/airnode-validator/test/fixtures
```
Comment (Siegrift) [inline .eslintignore:17]: We also do not need these lines.

File: .prettierignore [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+# https://github.com/protofire/solhint/issues/317
+#
+# /* solhint-disable */ comment doesn't work either because this is a parser issue
+# Prettier uses this parser for formatting
+packages/airnode-adapter/contracts/TestDecoder.sol
+
+receipt.json
+config.json
+!packages/airnode-examples/integrations/**/config.json
+
```
Comment (Siegrift) [inline .prettierignore:28]: Most of the lines here are unrelated to this project.

File: .solhintignore [hunk 8]
```diff
@@ -0,0 +1,7 @@
+**/node_modules/**
```
Comment (Siegrift) [inline .solhintignore:1]: Do we need this file?

File: src/aws-utils.ts [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+  });
+
+  const command = new DescribeLogGroupsCommand({});
+  const logGroups = (await client.send(command)).logGroups;
+
+  if (!logGroups?.filter) {
+    return;
+  }
+
+  const groups = logGroups.filter((group) => group!.logGroupName!.indexOf('airnode-') > -1);
```
Comment (Siegrift) [inline src/aws-utils.ts:39]:
```suggestion
  const groups = logGroups.filter((group) => group!.logGroupName!.includes('airnode-'));
```

File: src/aws-utils.ts [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+  const logStreams = await Promise.all(
+    (
+      await Promise.all(
```
Comment (Siegrift) [inline src/aws-utils.ts:42]: You can remove one layer of `Promise.all`

File: src/aws-utils.ts [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+  DescribeLogGroupsCommand,
+  DescribeLogStreamsCommand,
+  GetLogEventsCommand,
+  OutputLogEvent,
+} from '@aws-sdk/client-cloudwatch-logs';
+import { LogStream } from './types';
+import { contains } from './utils';
+import { getGreatestStats, compareDescendingOutputLogEvent } from './metrics-utils';
+
+/**
```
Comment (Siegrift) [inline src/aws-utils.ts:12]: 👍 Thanks for these comments

File: src/aws-utils.ts [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+
+  const groups = logGroups.filter((group) => group!.logGroupName!.indexOf('airnode-') > -1);
+  const logStreams = await Promise.all(
+    (
+      await Promise.all(
+        groups.map((group) => client.send(new DescribeLogStreamsCommand({ logGroupName: group.logGroupName })))
+      )
+    )
+      .map((ls) => ls.logStreams)
+      .filter((ls) => ls && ls!.length > 0)
```
Comment (Siegrift) [inline src/aws-utils.ts:47]:
```suggestion
      .filter((ls) => ls && ls.length > 0)
```

doesn't work?

File: src/aws-utils.ts [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+      ?.flatMap((ls) =>
```
Comment (Siegrift) [inline src/aws-utils.ts:48]: Does this need to be optional?

File: src/aws-utils.ts [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+      )
+      .map(async (ls) => {
+        const logData = await client.send(
+          new GetLogEventsCommand({
+            logStreamName: ls.logStreamName,
+            logGroupName: ls.logGroupName,
+          })
+        );
+
+        return { ...ls, logData: logData.events }; // there's a nextBackwardToken and nextForwardToken which may be needed
```
Comment (Siegrift) [inline src/aws-utils.ts:67]: What are these tokens?

File: src/aws-utils.ts [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+
+    if (!logs[ls.logGroupName]) {
+      logs[ls.logGroupName] = new Array<OutputLogEvent>();
+    }
+
+    logs[ls.logGroupName] = logs[ls.logGroupName].concat(ls.logData);
+  });
+
+  for (const [key, value] of Object.entries(logs)) {
+    // @ts-ignore
```
Comment (Siegrift) [inline src/aws-utils.ts:85]: Prefer `as any` to @ts-ignore

File: src/aws-utils.ts [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+    logs[key] = value.sort(compareDescendingOutputLogEvent); // ORDER BY DESC
+  }
+
+  const metrics = [];
+  for (const [key, value] of Object.entries(logs)) {
+    if (!value) {
```
Comment (Siegrift) [inline src/aws-utils.ts:91]: Nit: you can do oneline

File: src/chain.ts [hunk 17]
```diff
@@ -1,19 +1,59 @@
...
 
+export const doHardHatMine = async (provider: ethers.providers.JsonRpcProvider, testType: string) => {
+  if (testType === 'HardHatProvider') {
+    await provider.send('evm_mine', []).catch((e) => {
+      console.trace('Mine block failed', e);
+    });
+  }
+};
+
+export class RetryingProvider extends ethers.providers.JsonRpcProvider {
```
Comment (Siegrift) [inline src/chain.ts:26]: It would be nice to document why this is needed.

Comment (Siegrift) [reply src/chain.ts:26]: It would be even more nice to have an info how many times did you need to retry an operation.

File: src/gcp-utils.ts [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+// the project specified by the GCLOUD_PROJECT environment variable. See
+// https://googlecloudplatform.github.io/gcloud-node/#/docs/google-cloud/latest/guides/authentication
+
+/**
+ * Trims the output of the gcloud command.
+ *
+ * @param a The stdout output from the gcloud command
+ */
+const trimGCloudOutput = (a: string) => a.substring(1, a.length - 2);
+/**
```
Comment (Siegrift) [inline src/gcp-utils.ts:19]: Nit: \n

File: src/implementation.ts [hunk 19]
```diff
@@ -0,0 +1,288 @@
+// Explicitly copied for safety from airnode-admin
```
Comment (Siegrift) [inline src/implementation.ts:1]: Why did you not use the airnode-admin package?

**Review by Siegrift — APPROVED** [review]
👍 LGTM

The code is much readable now - I just left a few comments.

**Comment by aquarat** [comment]
> The code is much readable now - I just left a few comments.

Thanks for the comments :) I just saw them now, I'll work on your suggestions ASAP.

File: .eslintignore (hunk 5 above)
Comment (aquarat) [reply .eslintignore:3]: Cleaned up :+1:

File: src/aws-utils.ts (hunk 10 above)
Comment (aquarat) [reply src/aws-utils.ts:42]: It is unhappy if I do that :grimacing:

File: src/aws-utils.ts (hunk 11 above)
Comment (aquarat) [reply src/aws-utils.ts:12]: I will add more in future, I wasn't sure how useful they were.

File: src/implementation.ts (hunk 19 above)
Comment (aquarat) [reply src/implementation.ts:1]: I was having issues related to the protocol change. This was a fast, lazy and effective way of fixing those issues.
I'm happy to remove this file and import instead.

File: src/aws-utils.ts (hunk 13 above)
Comment (aquarat) [reply src/aws-utils.ts:48]: Apparently not hah, I've removed it.

File: src/aws-utils.ts (hunk 14 above)
Comment (aquarat) [reply src/aws-utils.ts:67]: They provide cursor functionality (like, the results are 3 "pages" long, to get the next page you provide the nextForwardToken in a follow-up call), but so far the API response has never been big enough for them to be required.

File: src/aws-utils.ts (hunk 15 above)
Comment (aquarat) [reply src/aws-utils.ts:85]: There was an upstream change that apparently made those @ts-ignore statements redundant. I've removed them :)

File: src/gcp-utils.ts (hunk 18 above)
Comment (aquarat) [reply src/gcp-utils.ts:20]: Fixed :+1:

File: src/chain.ts (hunk 17 above)
Comment (aquarat) [reply src/chain.ts:43]: Documentation added.
This is an issue that affects the stress tester when it does large batches of EVM calls, it doesn't affect the Airnode instance being tested, hence me not adding any telemetry around it.
